sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -297,12 +307,20 @@ case class AdaptiveSparkPlanExec(
         // plans are updated, we can clear the query stage list because at this point the two plans
         // are semantically and physically in sync again.
         val logicalPlan = replaceWithQueryStagesInLogicalPlan(currentLogicalPlan, stagesToReplace)
-        val (newPhysicalPlan, newLogicalPlan) = reOptimize(logicalPlan)
-        val origCost = costEvaluator.evaluateCost(currentPhysicalPlan)
-        val newCost = costEvaluator.evaluateCost(newPhysicalPlan)
-        if (newCost < origCost ||
-            (newCost == origCost && currentPhysicalPlan != newPhysicalPlan)) {
-          logOnLevel(s"Plan changed from $currentPhysicalPlan to $newPhysicalPlan")
+        val (newPhysicalPlans, newLogicalPlan) = reOptimize(logicalPlan)
+        val newPhysicalPlan =
+          (Seq(currentPhysicalPlan) ++ newPhysicalPlans)
```

Comment:
Let's make the rule clear:
1. If the cost is the same, we prefer to pick the new physical plan
2. If the new physical plans have the same cost, we prefer to first plan (that has no skew join optimization)

Reply:
I think the code can be
```
val preferredNewPhysicalPlan = ... // the plan with lowest cost and comes first
if (newCost < oldCost || ...) ...
```